### ISSUE TITLE
False positive Number arguments (1) to String#% mismatches expected fields (6).

### ISSUE THREAD

**Issue by obfuscoder** [bbatsov/rubocop#2307]
I got this code:

```ruby
data = [1, 2, 3, 4, 5, 6]
order = '%02d%03d%01d%02d%02d%03d' % data
```

Running rubocop yields this:

```
t.rb:2:36: W: Number arguments (1) to String#% mismatches expected fields (6).
order = '%02d%03d%01d%02d%02d%03d' % data
```

I expect to not see any warnings here.

I am running rubocop 0.34.2 on JRuby 1.7.20

**Comment by bbatsov** [comment]
//cc @edmz
